firedrake/slate/slac/tsfc_driver.py

```diff
@@ -47,10 +60,12 @@ def compile_terminal_form(tensor, prefix=None, tsfc_parameters=None):
     cxt_kernels = []
     for orig_it_type, integrals in transformed_integrals.items():
         subkernel_prefix = prefix + "%s_to_" % orig_it_type
-        kernels = tsfc_compile(Form(integrals),
+        form = Form(integrals)
+        kernels = tsfc_compile(form,
                                subkernel_prefix,
                                parameters=tsfc_parameters)
         cxt_k = ContextKernel(tensor=tensor,
+                              form_indices_map=dict(split_form(form)),
                               original_integral_type=orig_it_type,
```

Comment:
This stuff is some serious action at a distance.  Inside `tsfc_compile`, we call `split_form` and unwind some coefficient numbering from the split forms into the original expression.  That information is then immediately thrown away, because we don't need it any more.

Here you're sort of "recreating" it, by remembering the split forms in the `ContextKernel`.